addons/stock/views/stock_move_views.xml

```diff
@@ -348,7 +348,7 @@
                 <search string="Stock Moves">
                     <field name="origin" filter_domain="['|', '|', ('origin', 'ilike', self), ('name', 'ilike', self), ('picking_id', 'ilike', self)]" string="Reference"/>
                     <field name="product_id"/>
-                    <field name="name" string="Location" filter_domain="['|',('location_id', 'ilike', self),('location_dest_id', 'ilike', self)]"/>
+                    <field name="location_id" string="Location" filter_domain="['|',('location_id', 'ilike', self),('location_dest_id', 'ilike', self)]"/>
```

Comment:
I think we can copy what we do in the stock.move.line search view : 
https://github.com/odoo/odoo/blob/7e8ed048985e9b0996e66da31fc1c619ea26acd0/addons/stock/views/stock_move_line_views.xml#L107-L108
It has the groups, one filter for source, one for dest and the filter_domains are not really necessary as it searches on the full parent_path. 
Thank you